docs/design/caching_scan_results_by_repo_digest.md

```diff
@@ -0,0 +1,117 @@
+I1008 19:58:19.376902   62385 reflector.go:219] Starting reflector *v1.Job (30m0s) from pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167
+I1008 19:58:19.376937   62385 reflector.go:255] Listing and watching *v1.Job from pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167
+I1008 19:58:19.386049   62385 runnable_job.go:130] Event: Created pod: scan-vulnerabilityreport-cbf8c9b99-4nzkb (SuccessfulCreate)
+I1008 19:58:51.243554   62385 runnable_job.go:130] Event: Job completed (Completed)
+I1008 19:58:51.247251   62385 runnable_job.go:109] Stopping runnable job on task completion with status: Complete
+I1008 19:58:51.247273   62385 runner.go:83] Stopping runner on task completion with error: <nil>
+I1008 19:58:51.247278   62385 scanner.go:130] Scan job completed: starboard/scan-vulnerabilityreport-cbf8c9b99
+I1008 19:58:51.247297   62385 scanner.go:262] Getting logs for nginx container in job: starboard/scan-vulnerabilityreport-cbf8c9b99
+I1008 19:58:51.674449   62385 scanner.go:123] Deleting scan job: starboard/scan-vulnerabilityreport-cbf8c9b99
+```
+
+Now, if we scan the `nginx` Deployment in the `bar` namespace, Starboard will
+see that there's already a ClusterVulnerabilityReport (`84bcb5cd46`) for the
+same image reference `nginx:1.16` and will skip creation of a K8s Job. It will
+just read and copy the report as VulnerabilityReport object to the `bar`
+namespace.
+
+```console
+$ kubectl get clustervulnerabilityreports -o wide
+NAME         REPOSITORY      TAG    DIGEST   SCANNER   AGE   CRITICAL   HIGH   MEDIUM   LOW   UNKNOWN
+84bcb5cd46   library/nginx   1.16            Trivy     17s   21         50     33       104   0
+```
+
+```console
+$ starboard scan vulnerabilityreports deploy/nginx -n bar -v 3
+I1008 19:59:23.891718   62478 scanner.go:72] Getting Pod template for workload: {Deployment nginx bar}
+I1008 19:59:23.895310   62478 scanner.go:89] Checking if image nginx:1.16 was already scanned
+I1008 19:59:23.903058   62478 scanner.go:95] Cache hit
+I1008 19:59:23.903078   62478 scanner.go:97] Copying ClusterVulnerabilityReport to VulnerabilityReport
+```
+
+As you can see, Starboard eventually created two VulnerabilityReports by spinning
+up only one K8s Job.
+
+```console
+$ kubectl get vulnerabilityreports -A
+NAMESPACE   NAME                                REPOSITORY      TAG    SCANNER   AGE
+bar         replicaset-nginx-6d4cf56db6-nginx   library/nginx   1.16   Trivy     5m38s
+foo         replicaset-nginx-6d4cf56db6-nginx   library/nginx   1.16   Trivy     6m10s
+```
+
+## Summary
+
+* This solution might be the first step towards more efficient vulnerability scanning.
+* It's backward compatible and can be implemented as an experimental feature behind
+  a gate.
+* Both Starboard CLI and Starboard Operator can read and leverage ClusterVulnerabilityReports.
+
+## TODO
+
+* Specify the life-cycle of ClusterVulnerabilityReport objects, e.g. how they are invalidated.
```

Comment:
I assume the main goal is to delete the ClusterVulnerabilityReport after a specific time.

One way could be to follow the steps of: https://github.com/aquasecurity/starboard/pull/879 which is to have a seperate controller sitting and listening for ClusterVulnerabilityReport. But in this case we woulden't do any reque instead we would just delete the existing ClusterVulnerabilityReport after a specific amount of time.

The good thing with this solution is that it would be very similar to what we do in https://github.com/aquasecurity/starboard/pull/863.

The second way could be when checking if a job should be started or not vs copying an existing ClusterVulnerabilityReport into a VulnerabilityReports we could just check and see the age of the ClusterVulnerabilityReport. If it's above x time we could just delete the existing ClusterVulnerabilityReport and run the job as we normally would, thus generating a new VulnerabilityReports and ClusterVulnerabilityReport.

The good thing here is that we don't rely on another controller.